dev_guide/app_tutorials/jenkins_pipeline.adoc

```diff
@@ -0,0 +1,185 @@
+[[dev-guide-jenkins-pipeline-builds]]
+= Jenkins Pipeline Builds
+{product-author}
+{product-version}
+:data-uri:
+:icons:
+:experimental:
+:toc: macro
+:toc-title:
+:prewrap!:
+
+toc::[]
+
+[[introduction]]
+
+== Introduction
+
+Using the Jenkins OpenShift Client Plugin, you can quickly and easily create Jenkins Pipelines utilizing the OpenShift DSL
+to get fine grained control over the build, deployment, and promotion of applications on your OpenShift cluster.
+
+For this example, we are going to create a Jenkins Pipeline that will build, deploy, and verify a Node.js/MongoDB
+application using the https://github.com/openshift/nodejs-ex/blob/master/openshift/templates/nodejs-mongodb.json[nodejs-mongodb.json] template.
+
+[[creating-the-jenkins-master]]
+
+== Creating the Jenkins Master
+
+Creating the Jenkins master via the command line:
+
+----
+  // select the project that you want to use
+  // or create a new project with oc new-project <project_name>
+  $ oc project <project_name>
+  // if you want to use persistent storage, use jenkins-persistent instead
+  $ oc new-app jenkins-ephemeral
+----
+
+[NOTE]
+====
+If Jenkins auto-provisioning is enabled on your cluster, and you don't need to make any customizations to the
+Jenkins master, you can skip the above step. For more information about Jenkins autoprovisioning see xref:../../install_config/configuring_pipeline_execution.html[Configuring Pipeline Execution].
+====
+
+[[the-pipeline-build-config]]
+
+== The Pipeline Build Config
+
+Now that the Jenkins master is up and running, we will create a build config that employs the Jenkins pipeline strategy
+to build, deploy, and scale our Node.js/MongoDB example application.
+
+Create a file named *nodejs-sample-pipeline.yaml* with the below content:
+
+[source,yaml]
+----
+kind: "BuildConfig"
+apiVersion: "v1"
+metadata:
+  name: "nodejs-sample-pipeline"
+spec:
+  strategy:
+    jenkinsPipelineStrategy:
+      jenkinsfile: <pipeline content from below>
+    type: JenkinsPipeline
+----
+
+For more information about configuring the Pipeline Build Strategy see xref:../../builds/build_strategies.html#pipeline-strategy-options[Pipeline Strategy Options].
+
+[[the-jenkinsfile]]
+
+== The Jenkinsfile
+
+Now that we have created the build config with a jenkinsPipelineStrategy, we need to tell the pipeline what to do.
+We will do this by using an inline Jenkinsfile, since we aren't going to be creating our own git repository for this application.
```

Comment:
This is not a call for change, as I believe I understand the pros/cons of switching approaches, and am fine with this safer path, but I thought it at least noteworthy to point out that placing the Jenkinsfile in the repo in question is the "preferred Jenkins way", for lack of a better term, and more than likely the approach our users will take if their dev process uses Jenkins.

Something to monitor moving forward I think.

Reply:
I will update the doc mentioning that as the preferred way, but thought that the inline jenkinsfile would be better for the example/tutorial

Reply:
updated in next push

Reply:
Yep that is what I was assuming the thinking was, and as I tried to convey, I agree, certainly for this initial pass ... if anything else, just leaving some bread crumbs for if/when we revisit.